### PR TITLE
install: Add explicit RBAC permissions for finalizers subresources

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

File: install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml [hunk 1]
```diff
@@ -37,12 +37,14 @@ rules:
...
   resources:
   - pods
+  - pods/finalizers
   - nodes
   verbs:
   - get
   - list
   - watch
   - update
+  - delete
```
Comment (aanm) [inline install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml:46]: ~why do we need the delete?~

I just read the commit description. we only need to delete pods, not `nodes` so we should have a different rule for `pods` that accepts get, list, watch, update and delete verbs. and a `nodes` rule that doesn't have the `delete`.

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml (hunk 1 above)
Comment (errordeveloper) [reply install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml:46]: > why do we need the delete?

It's explained in the commit message :)

**Review by aanm — CHANGES_REQUESTED** [review]
all commits could be squashed into 1

File: install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml (hunk 1 above)
Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml:46]: @errordeveloper I've edited my message

Comment (errordeveloper) [reply install/kubernetes/cilium/charts/agent/templates/clusterrole.yaml:46]: @aanm thanks, I will refine the rules!

**Comment by errordeveloper** [comment]
> all commits could be squashed into 1

@aanm I was mostly thinking that cherry-picking could be easier this way, as not all changes will apply to 1.8, but first commit should apply cleanly - wdyt?

**Comment by aanm** [comment]
> > all commits could be squashed into 1
> 
> @aanm I was mostly thinking that cherry-picking could be easier this way, as not all changes will apply to 1.8, but first commit should apply cleanly - wdyt?

@errordeveloper won't we have to backport all commits to 1.8? also, we will regenerate the manifests after backporting the commits so the conflicts would be easy to fix.

**Comment by errordeveloper** [comment]
@aanm we would need to backport all commits indeed, I don't think it changes anything - will squash now.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by errordeveloper** [comment]
@aanm this looks like it's good to go?

**Review by tgraf — APPROVED** [review]

**Pull request merged**